core/service/session_manager.go

```diff
@@ -267,8 +280,14 @@ func (manager *SessionManager) startSession(session *Session, prices market.Pric
 }
 
 func (manager *SessionManager) validateSession(session *Session, prices market.Price) error {
-	if !manager.service.Policies().IsIdentityAllowed(session.ConsumerID) {
-		return fmt.Errorf("consumer identity is not allowed: %s", session.ConsumerID.Address)
+	if manager.service.PolicyOracle().PeriodicFetchingEnabled() {
+		if !manager.service.Policies().IsIdentityAllowed(session.ConsumerID) {
+			return fmt.Errorf("consumer identity is not allowed: %s", session.ConsumerID.Address)
+		}
+	} else {
+		if !manager.service.PolicyOracle().IsIdentityAllowed(session.ConsumerID) {
+			return fmt.Errorf("consumer identity is not allowed: %s", session.ConsumerID.Address)
+		}
```

Comment:
Both `policies` and `policyOracle` implement the same `IsIdentityAllowed` method. 
Can we just add a single interface into the service manager to avoid all this duplicated code?

Reply:
Good suggestion :+1: